lib/src/auth.dart

```diff
@@ -55,35 +50,55 @@ abstract class Authenticator {
     if (credentials == null) {
       return;
     }
-    String accessToken = credentials.accessToken;
-    Map<String, String> revokeAccess = new Map<String, String>();
-    revokeAccess[kTokenKey] = accessToken;
-    revokeAccess[kTokenTypeHintKey] = 'access_token';
+    List<Map> tokens = new List<Map>();
+    Map accessToken = {
+      kTokenKey: credentials.accessToken,
+      kTokenTypeHintKey: 'access_token',
+    };
+    tokens.add(accessToken);
+
+    if (credentials.refreshToken != null) {
+      Map refreshToken = {
+        kTokenKey: credentials.refreshToken,
+        kTokenTypeHintKey: 'refresh_token',
+      };
+      tokens.add(refreshToken);
+    }
 
-    // Retrieve the client ID and secret.
-    String clientId = _grant.identifier;
-    String clientSecret = _grant.secret;
+    for (int i = 0; i < tokens.length; i++) {
```

Comment:
Prefer for-in loops when the index isn't used for anything other than indexing the collection.
```dart
for (var token in tokens) {
  ...
}
```